main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1345,6 +1352,51 @@ object Lowering {
     Expression.Tuple(exps, tpe, pur, eff, loc)
   }
 
+  /**
+    * Returns a new `VarSym` for use in a let-binding.
+    *
+    * This function is called `mkLetSym` to avoid confusion with [[mkVarSym]].
+    */
+  private def mkLetSym(prefix: String, loc: SourceLocation)(implicit flix: Flix): Symbol.VarSym = {
+    val name = prefix + Flix.Delimiter + flix.genSym.freshId()
+    Symbol.freshVarSym(name, BoundBy.Let, loc)
+  }
+
+  /**
+    * Returns a tuple expression that is evaluated in parallel.
+    */
+  private def mkParTuple(tuple: Expression.Tuple)(implicit flix: Flix): Expression = {
+    val Expression.Tuple(elms, tpe, pur, eff, loc) = tuple
+    val chanSymsWithElms = elms.map(e => (mkLetSym("channel", e.loc.asSynthetic), e))
+
+    // make wait expressions
+    val resultTuple = Expression.Tuple(chanSymsWithElms.map {
+      case (sym, e) =>
+        val loc = e.loc.asSynthetic
+        Expression.GetChannel(
+          Expression.Var(sym, Type.mkChannel(e.tpe, loc), loc),
+          e.tpe, Type.Impure, e.eff, loc)
+    }, tpe, pur, eff, loc.asSynthetic)
+
+    // make spawn expressions
+    val spawns = chanSymsWithElms.foldRight(resultTuple: Expression) {
+      case ((sym, e), acc) =>
+        val loc = e.loc.asSynthetic
+        val e1 = Expression.Var(sym, Type.mkChannel(e.tpe, loc), loc)
+        val e2 = Expression.PutChannel(e1, e, Type.Unit, Type.Impure, Type.mkUnion(e.eff, e1.eff, loc), loc)
```

Comment:
High-level question. We introduce `Impure` here for an expression that may not have been `Impure` before. Do we do this elsewhere? Is there reason to be concerned?

Reply:
My thinking is that it is an Impure expression in itself, but the caller wraps it in a Cast expression which casts it to pure. This is okay since we've checked during the typer that the original expression was pure.